### PR TITLE
Greatly improve terminal features! Persistent terminals

### PR DESCRIPTION
**Pull request by G-Rowell**
Hi all,

(terms = NeoVim `:terminal`s)

This is a PR to greatly improve our term functionality.
This can be summarised as now having persistent terms!
I particularly aim to never leave NeoVim when working, so this helps a lot in that sense!

# Improvements:
- Persistent terms with
-- These can create as many terms as you wish (as they just generate a new term)
-- `<leader> + w` for a window term
-- `<leader> + v` for a vertical split term
-- `<leader> + h` for a horizontal split term
-- A newly created term will start in insert mode
- Escape term mode with `jk`
- Escape & hide term with `JK`
- Bring them back with a Telescope picker `terms`
-- `<leader> + W` (note, capital W, so technically `<leader> + <shift> + w`
-- This runs `:Telescope terms`
- Improved close buffer behavior generally (also enables hiding of terms to keep them persistent)

- this also removes the `ToggleTerm` plugin, so you should run `:PackerSync`

# Known issues / wishlist:
- Auto start in insert mode when recovering one with the picker
- Statusline appears in terms again, need to fix this
- Improve picker for 100 columns terminal windows like I run (can't see the term buffer)
- Recovering window terms doesn't always put it back on the bufferline, if you are in a split of a term it will go into that window

## *Tl;dr* monkey can type cmatrix in terms now!

### CODE REVIEW

**Comment by G-Rowell** [comment]
Don't merge until I give the okay, I may be working on fixes

**Comment by G-Rowell** [comment]
Tested & approved by @siduck76

**Comment by siduck76** [comment]
waiting for @Akianonymus 's review and test!

**Comment by xTerma** [comment]
[เปิดโหมดป้องกัน]()
